File: rightrx/src/main/java/org/rodneyparshall/rightrx/model/Review.java
package org.rodneyparshall.rightrx.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.rodneyparshall.rightrx.domain.User;
import org.rodneyparshall.rightrx.enumeration.Dosage;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long reviewId;

    @ManyToOne
    @JoinColumn(name = "userId", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "drugId", nullable = false)
    private Drug drug;

    private String reviewInfo;
    private Dosage dosage;
}

File: rightrxapp/src/app/model/review.ts
import { Drug } from "./drug";
import { User } from "./user";

export class Review{
    public id: number|null = null;
    public drug: Drug = new Drug;
    public user: User = new User;

    constructor() {
        this.drug = this.drug;
        this.user = this.user;

    }
}
